roles/development/files/bin/make-changelog.py

```diff
@@ -50,42 +53,84 @@ def get_tags():
     cmd = "git tag -l"
     tags = run(cmd)
     tags.sort(key=LooseVersion)
-    tags.append('develop')
+    if 'develop' in run('git branch'):
+        tags.append('develop')
     return list(reversed(tags))
 
 
 def get_commits(start, stop):
-    cmd = "git log {start}...{stop} --pretty=format:'%H %s' --reverse | cat"
+    cmd = "git log {start}...{stop} --pretty=format:'%H %ci %s' --reverse | cat"
     cmd = cmd.format(start=start, stop=stop)
     output = run(cmd)
-    commits = [line.split(None, 1) for line in output]
+    commits = []
+    for line in output:
+        line = line.split(None, 4)
+        commits.append([line[0], ' '.join(line[1:4]), line[4]])
     return commits
 
 
-def commit_url_for(username, project, slug):
+def commit_url_for_github(username, project, slug):
     template = "https://github.com/{username}/{project}/commit/{slug}"
     return template.format(username=username, project=project, slug=slug[:9])
 
 
-def pull_url_for(username, project, number):
+def commit_url_for_pagure(username, project, slug):
+    template = "https://pagure.io/{project}/{slug}"
+    return template.format(project=project, slug=slug[:9])
+
+
+def pull_url_for_github(username, project, number):
     template = "https://github.com/{username}/{project}/pull/{number}"
     return template.format(username=username, project=project, number=number)
 
 
-def get_pull_info(username, project, number):
+def pull_url_for_pagure(username, project, number):
+    template = "https://pagure.io/{project}/pull-request/{number}"
+    return template.format(project=project, number=number)
+
+
+def get_pull_info_github(username, project, number):
     template = 'https://api.github.com/repos/' + \
         '{username}/{project}/pulls/{number}'
     url = template.format(username=username, project=project, number=number)
     password = commands.getoutput('pass sites/github')
-    response = github_session.get(url, auth=('ralphbean', password))
+    response = requests_session.get(url, auth=('ralphbean', password))
     body = response.json()
     title = body['title']
     author = body['user']['login']
-    link = pull_url_for(username, project, number)
+    link = pull_url_for_github(username, project, number)
+    return title, author, link
+
+
+def get_pull_info_pagure(username, project, number):
+    template = 'https://pagure.io/api/0/{project}/pull-request/{number}'
+    url = template.format(project=project, number=number)
+    response = requests_session.get(url)
+    body = response.json()
+    title = body['title']
+    author = body['user']['name']
+    link = pull_url_for_pagure(None, project, number)
     return title, author, link
 
 
-def main(username, project, version):
+def get_pull_requests_pagure(project, commit):
```

Comment:
How about passing the date into this function as a parameter named ``since`` instead of passing a ``commit``?  I think this is part of why I find it so confusing.  I see this function take a commit and I just assume that it is going to return the one pull request *for that commit*... but the commit is just being used as a starting place.